packages/frontend/src/components/VisualizationConfigs/ChartConfigPanel/Legend/ReferenceLines.tsx

```diff
@@ -70,21 +70,35 @@ export const ReferenceLines: FC<Props> = ({ items, projectUuid }) => {
                     );
                     if (selectedSeries === undefined) return;
 
+                    const useAverage = updateValue === 'average';
+
                     const dataWithAxis = {
-                        name: updateLabel || 'Reference line',
-                        value: lineId,
+                        name: updateLabel,
+                        type: useAverage ? 'average' : undefined,
+                        uuid: lineId,
                         lineStyle: { color: updateColor },
-                        label: updateLabel ? { formatter: updateLabel } : {},
-                        xAxis: undefined,
-                        yAxis: undefined,
-                        [dirtyLayout?.xField === fieldId ? 'xAxis' : 'yAxis']:
-                            updateValue,
+                        label: updateLabel
+                            ? { formatter: updateLabel }
+                            : useAverage
+                            ? { formatter: 'ave: {c}' }
```

Comment:
avg instead of ave?

Reply:
Yea, that's the right idea. I was saving space, but that's a losing battle :D